halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2Service.java

```diff
@@ -423,7 +423,8 @@ default String buildContainer(
 
   default TemplatedResource getProbe(ServiceSettings settings, Integer initialDelaySeconds) {
     TemplatedResource probe;
-    if (StringUtils.isNotEmpty(settings.getHealthEndpoint())) {
+    if (StringUtils.isNotEmpty(settings.getHealthEndpoint())
```

Comment:
Any thoughts on improving readability by lifting up the TCP case so we can keep the flag closest to the logic it actually gates? Something like "if no health endpoint OR useTcpProbe flag is on, use the TCP probe"

Reply:
Updated!

Reply:
You are so fast!!

Reply:
updated the !notEmpty